Panic

```
goroutine 1 [running]:
github.com/kshedden/datareader.(*SAS7BDAT).processByteArrayWithData(0xc0000e4000, 0x2fe8, 0x90, 0xc0000ead20, 0x40b7c8)
        /afs/umich.edu/user/k/s/kshedden/go/src/github.com/kshedden/datareader/sas7bdat.go:1190 +0x6b3
github.com/kshedden/datareader.(*SAS7BDAT).readline(0xc0000e4000, 0x1f40, 0x1f40, 0xc000108000)
        /afs/umich.edu/user/k/s/kshedden/go/src/github.com/kshedden/datareader/sas7bdat.go:813 +0x31d
github.com/kshedden/datareader.(*SAS7BDAT).Read(0xc0000e4000, 0x3e8, 0x13, 0x13, 0x0, 0x0, 0x1d)
        /afs/umich.edu/user/k/s/kshedden/go/src/github.com/kshedden/datareader/sas7bdat.go:660 +0x389
main.doConversion(0x50e6c0, 0xc0000e4000)
        /afs/umich.edu/user/k/s/kshedden/go/src/github.com/kshedden/datareader/cmd/stattocsv/main.go:31 +0x94c
main.main()
        /afs/umich.edu/user/k/s/kshedden/go/src/github.com/kshedden/datareader/cmd/stattocsv/main.go:144 +0x24a
```

```
panic: runtime error: slice bounds out of range [:12408] with capacity 12288

goroutine 1 [running]:
github.com/kshedden/datareader.(*SAS7BDAT).processByteArrayWithData(0xc0000e4000, 0x2fe8, 0x90, 0x30, 0x4ea720)
        /home/mezimm00/go/src/github.com/kshedden/datareader/sas7bdat.go:1208 +0x73f
github.com/kshedden/datareader.(*SAS7BDAT).readline(0xc0000e4000, 0x1f40, 0x1f40, 0xc000122000)
        /home/mezimm00/go/src/github.com/kshedden/datareader/sas7bdat.go:823 +0x304
github.com/kshedden/datareader.(*SAS7BDAT).Read(0xc0000e4000, 0x3e8, 0x13, 0x13, 0x0, 0x0, 0x1d)
        /home/mezimm00/go/src/github.com/kshedden/datareader/sas7bdat.go:670 +0x33e
main.doConversion(0x515d20, 0xc0000e4000)
        /home/mezimm00/datareader/cmd/stattocsv/main.go:31 +0x95a
main.main()
        /home/mezimm00/datareader/cmd/stattocsv/main.go:144 +0x307
```

Using a go.mod file inside datareader/cmd/stattocsv forces using "this" repository on go build.  Doing that, the stacktrace is now:
```
panic: runtime error: slice bounds out of range [:12408] with capacity 12288

goroutine 1 [running]:
github.com/kshedden/datareader.(*SAS7BDAT).processByteArrayWithData(0xc00019e000, 0x2fe8, 0x90, 0x30, 0x4ea720)
        /home/mezimm00/datareader/sas7bdat.go:1208 +0x73f
github.com/kshedden/datareader.(*SAS7BDAT).readline(0xc00019e000, 0x1f40, 0x1f40, 0xc000126000)
        /home/mezimm00/datareader/sas7bdat.go:823 +0x304
github.com/kshedden/datareader.(*SAS7BDAT).Read(0xc00019e000, 0x3e8, 0x13, 0x13, 0x0, 0x0, 0x1d)
        /home/mezimm00/datareader/sas7bdat.go:670 +0x33e
main.doConversion(0x515e40, 0xc00019e000)
        /home/mezimm00/datareader/cmd/stattocsv/main.go:31 +0x95a
main.main()
        /home/mezimm00/datareader/cmd/stattocsv/main.go:144 +0x307
```

I can't provide the file due to data sensitivity and I'm not familiar with the actual format of the file, so I'm not sure if this is an issue with my data file or the reader, but I do know I can hack the reader to not panic :)

--- a/sas7bdat.go
+++ b/sas7bdat.go
@@ -1205,6 +1205,14 @@ func (sas *SAS7BDAT) processByteArrayWithData(offset, length int) error {
 			return err
 		}
 	} else {
+		if offset+length > len(sas.cachedPage) {
+			oldPage := sas.cachedPage
+			err, ok := sas.readNextPage()
+			if err != nil || !ok {
+				return fmt.Errorf("error reading next page - %w", err)
+			}
+			sas.cachedPage = append(oldPage, sas.cachedPage...)
+		}
 		source = sas.cachedPage[offset : offset+length]
 	}
 

